src/vm/ceeload.cpp

```diff
@@ -6461,6 +6461,36 @@ void Module::NotifyDebuggerUnload(AppDomain *pDomain)
 }
 
 #if !defined(CROSSGEN_COMPILE)
+using GetTokenForVTableEntry_t = mdToken(STDMETHODCALLTYPE*)(HMODULE module, BYTE**ppVTEntry);
+
+
+HMODULE GetIJWHostModule()
+{
+#if !defined(FEATURE_PAL)
+    static HMODULE ijwHostModule = nullptr;
+    if (ijwHostModule == nullptr)
+    {
+        ijwHostModule = CLRGetModuleHandle(W("ijwhost.dll"));
```

Comment:
We should talk about this kind of hard-coding. Also, will the `ijwhost` get an add ref?

Reply:
The ijwhost will not get an add-ref with `GetModuleHandle`.

Reply:
Per our discussion offline, I've re-implemented this method to resolve the module by reading the import tables in the PE and resolving the module handle to whichever module `_CorDllMain` is imported from.

Reply:
Why is this better?

This looks like load order problems factory - if somebody happens to load desktop IJW assembly first, the pointer will get initialized to point to mscoree and the modules that reference ijwhost and get loaded later will break.

Reply:
I guess you probably forgot to remove the caching of the entrypoint in the global static from `GetTokenGetterFromHostModule `.

Reply:
Yeah I forgot to remove the caching in `GetTokenGetterFromHostModule`. Fixing that now.